### PR TITLE
Comments for ticket. Full-featured 😎

### PR DESCRIPTION
**Pull request by AntonAL**
Comments UI is implemented as a schema-agnostic `Comments` component.

All CRUD operations are implemented:
- For updating, a `Typography.Text` component with an `editable` prop is used.
- For deleting a soft-delete technique is used. In place of deleted comment, a placeholder "Comment was deleted" will be displayed.

If current employee does't have a `canManageTicketComments` ability, a message is displayed instead of a comment form.

### CODE REVIEW

File: apps/condo/domains/common/components/containers/GlobalStyle.tsx [hunk 1]
```diff
@@ -237,6 +240,72 @@ const uploadControlCss = css`
...
+  .ant-input.white {
+    background: white;
+    &:focus, &:active {
+      background: white !important;
+    }
+  }
+`
+
+const commentCss = css`
+  .ant-comment {
```
Comment (pahaz) [inline apps/condo/domains/common/components/containers/GlobalStyle.tsx:253]: is it possible not to use global styles? you can wrap a component with style overrides.

File: apps/condo/lang/ru.json [hunk 2]
```diff
@@ -234,8 +234,6 @@
   "pages.condo.property.type.village": "Коттеджный поселок",
   "pages.condo.property.unit.Name": "Название",
   "pages.condo.property.warning.modal.HelpMessage": "Информация о доме не сохранится, если вы уйдёте со страницы. Сохраните изменения или продолжите редактировать информацию о доме, закрыв это сообщение.",
-  "pages.condo.property.warning.modal.AddressValidationErrorMsg": "Вы не выбрали сооружение или тип выбранного сооружения не поддерживается",
```
Comment (pahaz) [inline apps/condo/lang/ru.json:237]: translations is not synced

File: apps/condo/pages/ticket/[id]/index.tsx [hunk 3]
```diff
@@ -27,6 +28,10 @@ import { UploadFileStatus } from 'antd/lib/upload/interface'
 import { TicketChanges } from '@condo/domains/ticket/components/TicketChanges'
 import ActionBar from '@condo/domains/common/components/ActionBar'
 import { OrganizationRequired } from '@condo/domains/organization/components/OrganizationRequired'
+import { Comments } from '@condo/domains/common/components/Comments'
+import { Property } from '../../../domains/property/utils/clientSchema'
```
Comment (pahaz) [inline apps/condo/pages/ticket/[id]/index.tsx:32]:
```suggestion
import { Property } from '@condo/domains/property/utils/clientSchema'
```

File: apps/condo/pages/ticket/[id]/index.tsx [hunk 4]
```diff
@@ -205,6 +212,22 @@ const TicketIdPage = () => {
...
+    })
+    const updateComment = TicketComment.useUpdate({}, () => {})
+    const deleteComment = TicketComment.useSoftDelete({}, () => {})
+
+    const createCommentAction = TicketComment.useCreate({
+        ticket: id,
+        user: auth.user.id,
+    }, () => { refetchComments() })
+
+    const { link: { role } } = useOrganization()
```
Comment (pahaz) [inline apps/condo/pages/ticket/[id]/index.tsx:229]: role is not required for all users!!

File: apps/condo/pages/ticket/[id]/index.tsx [hunk 5]
```diff
@@ -231,139 +254,160 @@ const TicketIdPage = () => {
...
+                                        </Button>
+                                    </ActionBar>
+                                    <TicketChanges
+                                        loading={ticketChangesResult.loading}
+                                        items={ticketChangesResult.objs}
+                                        total={ticketChangesResult.count}
+                                    />
+                                </Row>
+                            </Col>
+                            <Col span={1}>
```
Comment (pahaz) [inline apps/condo/pages/ticket/[id]/index.tsx:391]: col without rows?

File: apps/condo/domains/common/components/Comments/index.tsx [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+  > * {
+    padding: 24px
+  }
+
+  display: flex;
+  flex-flow: column nowrap;
+  align-content: space-between;
+`
+const Head = styled.div`
+  border-bottom: solid thin #D9D9D9;
```
Comment (pahaz) [inline apps/condo/domains/common/components/Comments/index.tsx:34]: should we extract the color?

File: apps/condo/domains/common/components/Comments/CommentForm.tsx [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+  .ant-form-item-explain {
+    display: none;
+  }
+  textarea {
+    padding-right: 45px;
+  }
+`
+
+interface ICommentFormProps {
+    action?: (formValues) => Promise<any>
```
Comment (dkoviazin) [inline apps/condo/domains/common/components/Comments/CommentForm.tsx:26]: Don't look like it is optional

File: apps/condo/domains/common/components/Comments/index.tsx (hunk 6 above)
Comment (SavelevMatthew) [reply apps/condo/domains/common/components/Comments/index.tsx:34]: Can use **lightGrey** from `@condo/domains/common/constants/style.js`

File: apps/condo/domains/common/components/containers/GlobalStyle.tsx (hunk 1 above)
Comment (AntonAL) [reply apps/condo/domains/common/components/containers/GlobalStyle.tsx]: Moved into component scope in `b3768a18`

File: apps/condo/lang/ru.json (hunk 2 above)
Comment (AntonAL) [reply apps/condo/lang/ru.json]: It seems, like @MrFoxPro has resolved it in `025e9759`.

File: apps/condo/pages/ticket/[id]/index.tsx (hunk 5 above)
Comment (AntonAL) [reply apps/condo/pages/ticket/[id]/index.tsx:391]: `Col`, that You highlighted, does have parent `Row`

File: apps/condo/domains/common/components/Comments/index.tsx (hunk 6 above)
Comment (AntonAL) [reply apps/condo/domains/common/components/Comments/index.tsx]: Refactored in `41e9b707`

File: apps/condo/domains/common/components/containers/GlobalStyle.tsx [hunk 8]
```diff
@@ -237,6 +238,15 @@ const uploadControlCss = css`
   }
 `
 
+const inputControlCss = css`
+  .ant-input.white {
+    background: white;
```
[the remark this answers is not in the document]
Comment (AntonAL) [reply apps/condo/domains/common/components/containers/GlobalStyle.tsx]: Refactored in `41e9b707`

File: apps/condo/pages/ticket/[id]/index.tsx (hunk 3 above)
Comment (AntonAL) [reply apps/condo/pages/ticket/[id]/index.tsx]: Changed in `41e9b707`

File: apps/condo/domains/common/components/Comments/CommentForm.tsx (hunk 7 above)
Comment (AntonAL) [reply apps/condo/domains/common/components/Comments/CommentForm.tsx]: Changed in `41804d18`

File: apps/condo/pages/ticket/[id]/index.tsx (hunk 4 above)
Comment (AntonAL) [reply apps/condo/pages/ticket/[id]/index.tsx]: Changed in `632e5485`, `604f6fe9`, `6ee0420c`.

**Review by dkoviazin — APPROVED** [review]
Ok

**Review by pahaz — APPROVED** [review]

**Pull request merged**